# Bibliotecas

In [2]:
import os
import pandas as pd
import numpy as np
import scipy as sp
from scipy import signal
import timeit
from datetime import datetime
from sklearn.externals import joblib
import matplotlib.pyplot as plt
from sklearn.decomposition import PCA
from sklearn import preprocessing
from sklearn import utils

%matplotlib notebook

In [3]:
def HStacker(files_,t):
    
    #Lê dados do arquivo e remove janela
    Xc = np.load('DB_REPO/preproc/'+files_[3])
    Xc = Xc[:-49] #Exclui do STD para igualar o Hshape
    Xci = np.load('DB_REPO/preproc/'+files_[0])
    Xci = Xci[:-49]
    Xc = np.hstack((Xc.reshape(Xc.shape),Xci.reshape(Xci.shape)))
    
    #Aplica Hstack nos dados
    for j in range(1,len(files_)-1):
        Xci = np.load('DB_REPO/preproc/'+files_[j])
        Xc = np.hstack((Xc.reshape(Xc.shape),Xci.reshape(Xci.shape)))
          
    
    # Salving in file on the folder <DB_PKS/preproc/labeled>
    np.save('./DB_REPO/stacked/'+t+'_hs', Xc, allow_pickle=False)
    DATA_hs.append(t+'_hs.npy')

In [19]:
def VStacker(files_,t):
    
    #Lê dados do arquivo e remove janela
    Xc = np.load('DB_REPO/stacked/'+files_[0])
    
    #Aplica Hstack nos dados
    for j in range(0,len(files_)-1):
        Xci = np.load('DB_REPO/stacked/'+files_[j])
        Xc = np.vstack((Xc,Xci)) 
          
    
    # Salving in file on the folder <DB_PKS/preproc/labeled>
    np.save('./DB_REPO/stacked/'+t+'_vs', Xc, allow_pickle=False)
    DATA_vs.append(t+'_vs.npy')

In [5]:
def HStackerF(files_,t):
    
    #Lê dados do arquivo e remove janela
    Xc = np.load('DB_REPO/stacked/'+files_[5])
    Xc = Xc[:35853] #Exclui amostras para igualar tamanho e possibilitar o vstack
    
    #Aplica Hstack nos dados
    for j in range(0,len(files_)-1):
        Xci = np.load('DB_REPO/stacked/'+files_[j])
        Xci = Xci[:35853] #Exclui amostras para igualar tamanho e possibilitar o vstack
        Xc = np.hstack((Xc.reshape(Xc.shape),Xci.reshape(Xci.shape)))
          
    
    # Salving in file on the folder <DB_PKS/preproc/labeled>
    np.save('./DB_REPO/stacked/'+t+'_fs', Xc, allow_pickle=False)
    DATA_fs.append(t+'_fs.npy')

In [6]:
#Carregando arquivos
import commands

A = commands.getoutput('ls DB_REPO/preproc')
files_ = A.split('\n')
DATA_hs = []

for i in range(0,len(files_),4):
    HStacker(files_[i:i+4],files_[i:i+4][0].split('-')[0]+str(i/4))
    #print (files_[i:i+3],files_[0:3][0].split('-')[0]+str(i/3))

In [7]:
files_[0:][0].split('-')[0]

'Arthur Araujo Filho '

In [8]:
files_[0:4]

['Arthur Araujo Filho - Dia 17.11.18 as 16.12.34.edf_fft.npy',
 'Arthur Araujo Filho - Dia 17.11.18 as 16.12.34.edf_filt.npy',
 'Arthur Araujo Filho - Dia 17.11.18 as 16.12.34.edf_rms.npy',
 'Arthur Araujo Filho - Dia 17.11.18 as 16.12.34.edf_std.npy']

In [9]:
for i in range(0,len(files_),4):
    r = files_[i:i+4]
    
    print (((np.load('DB_REPO/preproc/'+r[0])).shape[0],(np.load('DB_REPO/preproc/'+r[1])).shape[0],
            (np.load('DB_REPO/preproc/'+r[2])).shape[0],(np.load('DB_REPO/preproc/'+r[3])).shape[0]))

(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(14000, 13951, 13951, 14000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(14000, 13951, 13951, 14000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(14000, 13951, 13951, 14000)
(12000, 11951, 11951, 12000)
(14000, 13951, 13951, 14000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(14000, 13951, 13951, 14000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951, 11951, 12000)
(12000, 11951,

In [26]:
import re 

print  (re.findall('\d*\D+',(DATA_hs[0:4][0]))[0])



Arthur Araujo Filho 


In [13]:
DATA_hs

['Arthur Araujo Filho 0_hs.npy',
 'Arthur Araujo Filho 1_hs.npy',
 'Arthur Araujo Filho 2_hs.npy',
 'Boanerges Ramos Cunha 3_hs.npy',
 'Boanerges Ramos Cunha 4_hs.npy',
 'Boanerges Ramos Cunha 5_hs.npy',
 'Carmino Pereira da Silva 6_hs.npy',
 'Carmino Pereira da Silva 7_hs.npy',
 'Carmino Pereira da Silva 8_hs.npy',
 'Diniz Esmeraldo 9_hs.npy',
 'Diniz Esmeraldo 10_hs.npy',
 'Diniz Esmeraldo 11_hs.npy',
 'Donia Aires 12_hs.npy',
 'Donia Aires 13_hs.npy',
 'Donia Aires 14_hs.npy',
 'Fernando Francisco da Silva 15_hs.npy',
 'Fernando Francisco da Silva 16_hs.npy',
 'Fernando Francisco da Silva 17_hs.npy',
 'Gastao Goncalves da Silva 18_hs.npy',
 'Gastao Goncalves da Silva 19_hs.npy',
 'Gastao Goncalves da Silva 20_hs.npy',
 'Ismael Rodrigues do Carmo 21_hs.npy',
 'Ismael Rodrigues do Carmo 22_hs.npy',
 'Ismael Rodrigues do Carmo 23_hs.npy',
 'Ivanildes Barbosa 24_hs.npy',
 'Ivanildes Barbosa 25_hs.npy',
 'Ivanildes Barbosa 26_hs.npy',
 'Jose Fernandes Araujo 27_hs.npy',
 'Jose Fernandes 

In [14]:
Xc = np.load('DB_REPO/stacked/Arthur Araujo Filho 0_hs.npy')

In [15]:
dfn = pd.DataFrame(data=Xc)
dfn.head()

0         1         2         3             4             5   \
0  0.022327 -0.001221  0.186603  0.017612  83175.300000  31503.679104   
1  0.019318 -0.001221  0.186603  0.015084  35594.691359  12367.931209   
2  0.025335  0.000705  0.173561  0.024354  57290.183123   5324.741546   
3  0.016310 -0.000258  0.173561  0.013398  36185.811721   7167.004118   
4  0.022327  0.000224  0.199644  0.015926  22033.382078   4793.733812   

             6             7         8         9         10        11  \
0  64245.622857  85142.152672  0.177580  0.159890 -0.286540  0.051892   
1  44455.620186  39285.893634  0.137408  0.217005 -0.212902 -0.126882   
2  89391.768584  35538.258799  0.147789  0.318284 -0.122926 -0.406303   
3  26367.619636  56738.077665  0.282882  0.487074 -0.007218 -0.741849   
4  28252.036321  34124.670679  0.552582  0.706690  0.125195 -0.980704   

          12         13         14         15  
0  11.446258  13.161945  24.856702  19.817440  
1  11.403695  13.155661  24.961891  19.516842  
2  11.341954  13.177480  25.011108  19.332502  
3  11.341954  13.261443  25.167758  18.626781  
4  11.657292  13.429970  25.293589  18.562591

In [16]:
Xc.shape

(11951, 16)

In [17]:
len(DATA_hs)

42

In [20]:
#Carregando arquivos
DATA_vs = []

for i in range(0,len(DATA_hs),3):
    VStacker(DATA_hs[i:i+3],(re.findall('\d*\D+',DATA_hs[i:i+3][0])[0]+str(i/3)))

In [21]:
len(DATA_vs)

14

In [24]:
Xc = np.load('DB_REPO/stacked/Arthur Araujo Filho 0_vs.npy')
dfn = pd.DataFrame(data=Xc)
#dfn.head()

Xc[:35853].shape

(35853, 16)

for i in range(0,len(DATA_vs),1):
    print min(np.load('DB_REPO/stacked/'+DATA_vs[i:i+1][0]).shape[0],np.load('DB_REPO/stacked/'+DATA_vs[i:i+1][1]).shape[0],
              np.load('DB_REPO/stacked/'+DATA_vs[i:i+1][2]).shape[0],np.load('DB_REPO/stacked/'+DATA_vs[i:i+1][3]).shape[0],
              np.load('DB_REPO/stacked/'+DATA_vs[i:i+1][4]).shape[0], np.load('DB_REPO/stacked/'+DATA_vs[i:i+1][5]).shape[0])

i= 6
print min(np.load('DB_REPO/stacked/'+DATA_vs[i:i+6][0]).shape[0],np.load('DB_REPO/stacked/'+DATA_vs[i:i+6][1]).shape[0],
           np.load('DB_REPO/stacked/'+DATA_vs[i:i+6][2]).shape[0],np.load('DB_REPO/stacked/'+DATA_vs[i:i+6][3]).shape[0],
           np.load('DB_REPO/stacked/'+DATA_vs[i:i+6][4]).shape[0], np.load('DB_REPO/stacked/'+DATA_vs[i:i+6][5]).shape[0])

In [27]:
DATA_vs

['Arthur Araujo Filho 0_vs.npy',
 'Boanerges Ramos Cunha 1_vs.npy',
 'Carmino Pereira da Silva 2_vs.npy',
 'Diniz Esmeraldo 3_vs.npy',
 'Donia Aires 4_vs.npy',
 'Fernando Francisco da Silva 5_vs.npy',
 'Gastao Goncalves da Silva 6_vs.npy',
 'Ismael Rodrigues do Carmo 7_vs.npy',
 'Ivanildes Barbosa 8_vs.npy',
 'Jose Fernandes Araujo 9_vs.npy',
 'Jose Francisto Fontineli  10_vs.npy',
 'Marcilio Pontinho de Rezende 11_vs.npy',
 'Osvaldir Borborema 12_vs.npy',
 'Sergio Lamenza 13_vs.npy']

DATA_vs[78:78+6]

DATA_fs = []

for i in range(0,len(DATA_vs),6):
    HStackerF(DATA_vs[i:i+6],(re.findall('\d*\D+',DATA_vs[i:i+6][0])[0]+str(i/6)))

DATA_fs

In [45]:
def proccess_data(x,DATA,LABEL):
    
    Xc = np.load("DB_REPO/stacked/"+x)
    print '\nXc initial shape ', Xc.shape
    
    #Principal component analysis
    pca = PCA(n_components=12)
    pca.fit(Xc)
    Xc = pca.transform(Xc)
    print pca.explained_variance_ratio_
    print ('Entropia: %3f' % (sum(pca.explained_variance_ratio_)))
    print '\nXc PCA shape ', Xc.shape
    
    #Create temporal serie
    #Xc = mls.generate_envelope(Xc, NUM_SAMPLES)
    #print 'Xc temporal-serie shape ', Xc.shape
    
    #Labeling the PKS level    
    C = (np.ones(len(Xc))*LABEL).reshape((len(Xc),1))
    Xc = np.hstack((Xc.reshape(Xc.shape),C.reshape((len(Xc),1)))) 
    print 'Xc labeled shape ', Xc.shape
    
    # Salving in file on the folder <DB_PKS/preproc/labeled>
    np.save('./DB_REPO/labeled/'+x[:2]+'_tsl', Xc, allow_pickle=False)
    print ''+x[:2]+'_tsl'
    
    DATA.append(x[:2]+'_tsl.npy')

---

#Validação
start = timeit.default_timer()

NUM_SAMPLES = 500
l = [1]
DATA = []

for x, y in zip(DATA_hs, l):
    proccess_data(x,DATA,NUM_SAMPLES,y)
    
stop = timeit.default_timer()
ti = stop - start 
print ('Tempo total para o envelopamento: %.2f min' % (ti/60))

DATA

Xc = np.load('DB_INTENC/preproc/labeled/p4_hs.npy_tsl.npy')
dfn = pd.DataFrame(data=Xc)
dfn.head()

---

In [46]:
#X = 'F'
#Y = 'F'
#Z = 'F'
#l = [1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 15, 2, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1,
# X, X, X, X, X, X, 15, 15, 15, 15, 15, 15, Y, Y, Y, Y, Y, Y, 15, 15, 15, 15, 15, 15, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
# 3, 3, 3, 3, 3, 3, Z, Z, Z, Z, Z, Z, 1, 1, 1, 1, 1, 1, 4, 4, 4, 4, 4, 4, 1, 1, 1, 1, 1, 1]

l = [1, 2, 15, 15, 2, 1, 15, 15, 1, 1, 3, 1, 4, 1]

#l = [2, 15, 4, 1, 0, 0, 15]
for x, y in zip(DATA_vs, l):
    print x[:-4],y

Arthur Araujo Filho 0_vs 1
Boanerges Ramos Cunha 1_vs 2
Carmino Pereira da Silva 2_vs 15
Diniz Esmeraldo 3_vs 15
Donia Aires 4_vs 2
Fernando Francisco da Silva 5_vs 1
Gastao Goncalves da Silva 6_vs 15
Ismael Rodrigues do Carmo 7_vs 15
Ivanildes Barbosa 8_vs 1
Jose Fernandes Araujo 9_vs 1
Jose Francisto Fontineli  10_vs 3
Marcilio Pontinho de Rezende 11_vs 1
Osvaldir Borborema 12_vs 4
Sergio Lamenza 13_vs 1


In [47]:
#Grupo total
start = timeit.default_timer()

DATA = []

for x, y in zip(DATA_vs, l):
    proccess_data(x,DATA,y)
    
stop = timeit.default_timer()
ti = stop - start 
print ('Tempo total para o envelopamento: %.2f min' % (ti/60))


Xc initial shape  (35853, 16)
[  8.05734026e-01   8.14866087e-02   6.32310722e-02   4.94984519e-02
   2.11570451e-05   1.66235600e-05   1.01982632e-05   1.49014884e-06
   1.13058288e-07   1.02227488e-07   9.56532301e-08   5.79183169e-08]
Entropia: 1.000000

Xc PCA shape  (35853, 12)
Xc labeled shape  (35853, 13)
Ar_tsl

Xc initial shape  (35853, 16)
[  6.76110564e-01   1.67747005e-01   1.25130850e-01   3.09760376e-02
   2.81249729e-05   3.64905668e-06   2.94115373e-06   4.26097122e-07
   1.85984710e-07   1.36297912e-07   4.33934967e-08   2.89245924e-08]
Entropia: 1.000000

Xc PCA shape  (35853, 12)
Xc labeled shape  (35853, 13)
Bo_tsl

Xc initial shape  (39853, 16)
[  7.58279586e-01   1.22848201e-01   7.20199220e-02   4.68356016e-02
   9.44226350e-06   3.65585877e-06   2.22313530e-06   8.21713138e-07
   1.98444124e-07   1.56161366e-07   9.93122359e-08   9.06495182e-08]
Entropia: 1.000000

Xc PCA shape  (39853, 12)
Xc labeled shape  (39853, 13)
Ca_tsl

Xc initial shape  (35853, 16)
[  

#Não foram incluidas as amostras p4 e c3, serão utilizadas para validação!

In [48]:
DATA[1:-2]

['Bo_tsl.npy',
 'Ca_tsl.npy',
 'Di_tsl.npy',
 'Do_tsl.npy',
 'Fe_tsl.npy',
 'Ga_tsl.npy',
 'Is_tsl.npy',
 'Iv_tsl.npy',
 'Jo_tsl.npy',
 'Jo_tsl.npy',
 'Ma_tsl.npy']

---

## Shape data

In [49]:
Xc = np.load("DB_REPO/labeled/"+DATA[0])
for i in DATA[1:-2]:
    Xc = np.vstack((Xc,np.load('DB_REPO/labeled/'+i))) 

### Split tipo 1

#Dividindo os dados em sets de treino, test e validação
dfn = pd.DataFrame(data=Xc)
train, validate, test = np.split(dfn.sample(frac=1), [int(.6*len(dfn)), int(.8*len(dfn))])

np.save('./DB_INTENC/preproc/labeled/train', train, allow_pickle=False)
np.save('./DB_INTENC/preproc/labeled/validate', validate, allow_pickle=False)
np.save('./DB_INTENC/preproc/labeled/test', test, allow_pickle=False)

Xc_train = np.load("DB_INTENC/preproc/labeled/train.npy")

X_train = Xc_train[:,0:(Xc_train.shape[1]-1)]

yz = Xc_train[:,[(Xc_train.shape[1]-1)]]
y_train = np.array([])
for i in range(len(yz)):
    y_train = np.hstack((y_train,yz[i]))
    
lab_enc = preprocessing.LabelEncoder()
y_train = lab_enc.fit_transform(y_train)

X_train.shape, y_train.shape

#Xc_validate = np.load("DB_INTENC/preproc/labeled/validate.npy")
Xc_validate = Xc

X_validate = Xc_validate[:,0:(Xc_validate.shape[1]-1)]

yz = Xc_validate[:,[(Xc_validate.shape[1]-1)]]
y_validate = np.array([])
for i in range(len(yz)):
    y_validate = np.hstack((y_validate,yz[i]))
    
#lab_enc = preprocessing.LabelEncoder()
#y_validate = lab_enc.fit_transform(y_validate)

X_validate.shape, y_validate.shape

Xc_test= np.load("DB_INTENC/preproc/labeled/validate.npy")

X_test = Xc_test[:,0:(Xc_test.shape[1]-1)]

yz = Xc_test[:,[(Xc_test.shape[1]-1)]]
y_test = np.array([])
for i in range(len(yz)):
    y_test = np.hstack((y_test,yz[i]))
    
lab_enc = preprocessing.LabelEncoder()
y_test = lab_enc.fit_transform(y_test)

X_test.shape, y_test.shape

### Split tipo 2

In [50]:
DATA[2]

'Ca_tsl.npy'

In [51]:
for i in DATA[1:3]:
    print i

Bo_tsl.npy
Ca_tsl.npy


In [52]:
#Separando set de dados X, e set de labels y - Validação DATA[5] e DATA[6]
Xc = np.load("DB_REPO/labeled/"+DATA[0])

for i in DATA[1:3]:
    Xc = np.vstack((Xc,np.load('DB_REPO/labeled/'+i))) 
    
X_validate = Xc[:,0:(Xc.shape[1]-1)]

yz = Xc[:,[(Xc.shape[1]-1)]]
y_validate = np.array([])
for i in range(len(yz)):
    y_validate = np.hstack((y_validate,yz[i]))
    
X_validate.shape, y_validate.shape

((111559, 12), (111559,))

In [61]:
X_validate_std = sc.transform(X_validate)

X_validate_std.shape, y_validate.shape

((111559, 12), (111559,))

In [57]:
#Separando set de dados X, e set de labels y - Treino e Teste
Xc = np.load("DB_REPO/labeled/"+DATA[3])
#for i in DATA[1:-2]:
for i in DATA[4:]:
    Xc = np.vstack((Xc,np.load('DB_REPO/labeled/'+i))) 
    
X = Xc[:,0:(Xc.shape[1]-1)]

yz = Xc[:,[(Xc.shape[1]-1)]]
y = np.array([])
for i in range(len(yz)):
    y = np.hstack((y,yz[i]))
    
X.shape, y.shape

((398383, 12), (398383,))

In [58]:
np.unique(y)

array([  1.,   2.,   3.,   4.,  15.])

In [59]:
#Dividindo os dados em sets de treino e test
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=1)

#X_test, X_validate, y_test, y_validate = train_test_split(X, y, test_size=0.5, random_state=1)

In [60]:
#Aplica normalizador P.S.: Verificar necessidade pois os dados já foram normalizados anteriormente
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
sc.fit(X_train)
X_train_std = sc.transform(X_train)
X_test_std = sc.transform(X_test)
X_validate_std = sc.transform(X_validate)

print X_train_std.shape, X_test_std.shape, X_validate_std.shape, y_train.shape, y_test.shape, y_validate.shape

(278868, 12) (119515, 12) (111559, 12) (278868,) (119515,) (111559,)


---

## Classifiers

### Random Forest

In [62]:
#Training and test
from sklearn.ensemble import RandomForestClassifier
rfc = RandomForestClassifier(n_estimators=100, n_jobs = 4)

print 'Training time'
%time rfc = rfc.fit(X_train_std, y_train)

print''

print 'Predict time'
%time y_pred = rfc.predict(X_test_std)

print''

from sklearn.metrics import accuracy_score
print ('ClassifyRF accuracy:---------->%.2f %%' % (accuracy_score(y_test, y_pred)*100))

Training time
CPU times: user 3min 55s, sys: 848 ms, total: 3min 56s
Wall time: 1min 3s

Predict time
CPU times: user 5.86 s, sys: 48 ms, total: 5.91 s
Wall time: 1.63 s

ClassifyRF accuracy:---------->97.34 %


### Memória anterior
Training time
CPU times: user 4min 12s, sys: 1.13 s, total: 4min 13s
Wall time: 1min 20s

Predict time
CPU times: user 3.82 s, sys: 188 ms, total: 4 s
Wall time: 1.45 s

ClassifyRF accuracy:---------->98.35 %


#Salva a rede treinada para um arquivo
joblib.dump(rfc, 'DB_INTENC/preproc/labeled/rfc_split2.pkl') 

rfc = joblib.load('DB_INTENC/preproc/labeled/rfc.pkl')

In [63]:
#Validation
print 'Predict time'
%time y_pred = rfc.predict(X_validate)

print''

from sklearn.metrics import accuracy_score

result = ('ClassifyRF accuracy:---------->%.2f %%' % (accuracy_score(y_pred, y_validate)*100))

print result

Predict time
CPU times: user 1.53 s, sys: 52 ms, total: 1.58 s
Wall time: 543 ms

ClassifyRF accuracy:---------->35.96 %


### Memória anterior
Predict time
CPU times: user 1.37 s, sys: 216 ms, total: 1.59 s
Wall time: 591 ms

ClassifyRF accuracy:---------->22.04 %


In [64]:
#Validation
print 'Predict time'
%time y_pred = rfc.predict(X_validate_std)

print''

from sklearn.metrics import accuracy_score

result = ('ClassifyRF accuracy:---------->%.2f %%' % (accuracy_score(y_pred, y_validate)*100))

print result

Predict time
CPU times: user 3.22 s, sys: 8 ms, total: 3.22 s
Wall time: 932 ms

ClassifyRF accuracy:---------->7.13 %


### Memória anterior
Predict time
CPU times: user 2.17 s, sys: 104 ms, total: 2.27 s
Wall time: 810 ms

ClassifyRF accuracy:---------->18.04 %

---

## Support Vector Machine

In [ ]:
from sklearn import svm, datasets

print 'Processing time SVM'
%time svm = svm.SVC(kernel='rbf', gamma=0.2, C=6, decision_function_shape='ovr').fit(X_train_std, y_train)

print ''

from sklearn.metrics import accuracy_score

print 'Predict time'  
%time y_pred = svm.predict(X_test_std)
from sklearn.metrics import accuracy_score
print ('SVM-RBF accuracy:---------->%.2f %%' % (accuracy_score(y_test, y_pred)*100))

Processing time SVM


In [54]:
print 'Predict time'  
%time y_pred = svm.predict(X_validate_std)
from sklearn.metrics import accuracy_score
print ('SVM-RBF accuracy:---------->%.2f %%' % (accuracy_score(y_validate, y_pred)*100))

Predict time
CPU times: user 7min 15s, sys: 436 ms, total: 7min 15s
Wall time: 7min 16s
SVM-RBF accuracy:---------->59.99 %
